src/rust-2021/prelude.md

```diff
@@ -43,6 +43,47 @@ In order to have rustfix migrate your code to be Rust 2021 Edition compatible, r
 cargo fix --edition
 ```
 
+The lint detects cases where functions or methods are called that have the same name as the methods defined in one of the new prelude traits. In some cases, it may rewrite your calls in various ways to ensure that you continue to call the same function you did before:
+
+| Scenario                                                                 | Example               | Rewrite (if any)     |
+| ------------------------------------------------------------------------ | --------------------- | -------------------- |
+| [Fully qualified inherent](#fully-qualified-calls-to-inherent-methods)   | `Foo::from_iter(...)` | none                 |
+| [Inherent methods on `dyn Trait`](inherent-methods-on-dyn-trait-objects) | `foo.into_iter()`     | `(*foo).into_iter`() |
+
+### Fully qualified calls to inherent methods
+
+Many types define their own inherent methods with the name `from_iter`:
+
+```rust
+struct Foo {
+  data: Vec<u32>
+}
+impl Foo {
+  fn from_iter(x: impl Iterator<Item = u32>) -> Self {
+    Foo {
+      data: x.collect()
+    }
+  }
+}
+```
+
+Calls like `Foo::from_iter` cannot be confused with calls to `<Foo as FromIter>::from_iter`, because inherent methods take precedence over trait methods.
```

Comment:
I think also we should connect the dots for the reader. Something like: "rewriting method calls to use fully qualified syntax ensures that the inherent method will be called".